ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/DivulgenceIT.scala

```diff
@@ -223,4 +230,37 @@ final class DivulgenceIT extends LedgerTestSuite {
       // nothing to test, if the workflow ends successfully the test is considered successful
     }
   })
+
+  test(
+    "DivulgenceDivulgeAfterArchival",
+    "Divulging, archiving and divulging again a contract with key should be possible",
+    allocate(SingleParty, SingleParty),
+  )(implicit ec => { case Participants(Participant(alpha, partyA), Participant(beta, partyB)) =>
+    for {
+      // Create a helper contract where partyB is a signatory
+      helper <- beta.create(
+        partyB,
+        WithKeyDivulgenceHelper(divulgedTo = partyB, withKeyOwner = partyA),
+      )
+
+      _ <- synchronize(alpha, beta)
+
+      // Create a WithKey contract owned by the partyA
+      withKey1 <- alpha.create(partyA, WithKey(partyA))
+
+      // Archive the withKey1 contract
+      _ <- alpha.exercise(partyA, helper.exerciseWithKeyDivulgenceHelper_Archive(_, withKey1))
```

Comment:
I’m not quite sure this is testing what you want to test. My understanding is we want to test the following (for simplicity assume each party has their own participant so party = participant).
1. A creates a contract where they are the only stakeholder.
2. A divulges that contract to B.
3. A archives that contract (without B knowing about the archive).
4. A creates another contract with the same key.
5. A divulges that contract to B.

You mixed the divulgence with the archival which means that even if B didn’t track keys for divulged contracts properly, it would never see a state where both contracts are archived.

Reply:
Do you mean that combining divulgence and archiving contract in the same exercise will work differently than doing this in two steps?

Reply:
Yes in my scenario B does not see the archive. That was exactly the bug around disclosure: B saw two contracts without an interleaving archive.
In your example, B does see the archive so it never has two contracts active at the same time.

Reply:
You're right! Thank you for catching this one.

Reply:
I've updated the test case. Please review when you have time.

Reply:
It seems like you are still divulging the archive. I think what you want is something like the following:
```
_ <- alpha.exercise(partyA, withKey1.exerciseArchive(_))
```

Reply:
🤦  Right. A fix incoming.